tests/PHPStan/Analyser/data/bug-instanceof-static-vs-this-prop-assign-scope.php

```diff
@@ -0,0 +1,30 @@
+<?php declare(strict_types=1);
+
+namespace BugInstanceofStaticVsThisPropAssignScope;
+
+class FooBase
+{
+	public function run(): void
+	{
+		if ($this instanceof FooChild) {
+			$this::$prop = 5;
+			\PHPStan\Testing\assertType('5', $this::$prop);
```

Comment:
Ideally, this should work even if the write is via `$this::` and the read is via `static::`, and vice-versa, but for now it doesn't.